lib/acpi/acpi.c

```diff
@@ -791,21 +791,29 @@ int acpi_drhd_get(enum AcpiDmarScopeType scope, struct acpi_dmar_device_scope *d
 	return 0;
 }
 
-struct acpi_madt_local_apic *acpi_local_apic_get(uint32_t cpu_num)
+#define ACPI_CPU_FLAGS_ENABLED 0x01u
+
+struct acpi_madt_local_apic *acpi_local_apic_get(int cpu_num)
 {
 	struct acpi_madt_local_apic *lapic;
 	int cpu_cnt;
+	int idx;
 
 	if (acpi_madt_entry_get(ACPI_MADT_TYPE_LOCAL_APIC, (ACPI_SUBTABLE_HEADER **)&lapic,
 				&cpu_cnt)) {
 		/* Error on MAD table. */
 		return NULL;
 	}
 
-	if ((cpu_num >= cpu_cnt) || !(lapic[cpu_num].LapicFlags & 1u)) {
-		/* Proccessor not enabled. */
-		return NULL;
+	for (idx = 0; cpu_num >= 0 && idx < cpu_cnt; idx++) {
+		if (lapic[idx].LapicFlags & ACPI_CPU_FLAGS_ENABLED) {
+			if (cpu_num == 0) {
```

Comment:
why you check for BSP only ? This might fail in case of RPL-S where the hard coded x86_cpu_loapics not same.

Reply:
Why would it fail? We update the x86_cpu_loapics in arch_start_cpu()